backend/src/utils/functional.js

```
@@ -0,0 +1,79 @@
+// --- ARRAY FUNCTIONS ---
+
+export const mapOver = curry((fn, arr) => arr.map(fn));
+export const filterBy = curry((predicate, arr) => arr.filter(predicate));
+export const reduceWith = curry((reducer, initial, arr) =>
+   arr.reduce(reducer, initial)
+);
+
+// --- OBJECT FUNCTIONS ---
+
+export const pick = curry((keys, obj) =>
+   keys.reduce((acc, key) => ({ ...acc, [key]: obj[key] }), {})
```

Biome (1.9.4)

[error] Avoid the use of spread (`...`) syntax on accumulators.

Spread syntax should be avoided on accumulators (like those in `.reduce`) because it causes a time complexity of `O(n^2)`.
Consider methods such as .splice or .push instead.

(lint/performance/noAccumulatingSpread)